src/fpm_compiler.f90

```diff
@@ -775,6 +816,7 @@ subroutine new_compiler(self, fc, cc, echo, verbose)
       self%cc = cc
     else
       call get_default_c_compiler(self%fc, self%cc)
+      call get_default_cpp_compiler(self%fc, self%cxx)
     end if
```

Comment:
Should the added not be in another if condition (as `len_trim(cc)` might be = 0 while `len_trim(cxx)` being > 0? 
```suggestion
      self%cc = cc
    else
      call get_default_c_compiler(self%fc, self%cc)
    end if
    if (len_trim(cxx) > 0) then
      self%cxx = cxx
    else
      call get_default_cpp_compiler(self%fc, self%cxx)
    end if
```